addons/website_sale/static/src/js/website_sale.js

```diff
@@ -350,6 +353,46 @@ export const WebsiteSale = publicWidget.Widget.extend(VariantMixin, {
             );  // Trigger `onChangeVariant` through .js_main_product
         }
     },
+    /**
+     * Search attribute values based on the input text.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _searchAttributeValues(ev) {
+        const input = ev.target;
+        const searchValue = input.value.toLowerCase();
+
+        document.querySelectorAll(`#${input.dataset.containerId} .form-check`).forEach(item =>{
+            const labelText = item.querySelector('.form-check-label').textContent.toLowerCase();
+            item.style.display = labelText.includes(searchValue) ? '' : 'none'
+        });
+    },
+    /**
+     * Highlight selected pill
+     *
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickPillsAttribute(ev) {
+        if (ev.target.tagName === "LABEL" || ev.target.tagName === "INPUT") {
+            return;
+        }
+        const checkbox = ev.target.closest('.o_wsale_variant_pills_shop').querySelector("input");
+        checkbox.click();
+    },
+    /**
+     * Toggle the button text between "View More" and "View Less"
+     *
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onToggleViewMore(ev) {
```

Comment:
nitpicking: So it's easier to understand what to expect from the function :)
```suggestion
    _onToggleViewMoreLabel(ev) {
```